### ISSUE TITLE
Improve documentation for handling removal of microphone access on iOS.

### ISSUE THREAD

**Issue by Zurigan** [ryanheise/just_audio#1007]
**To which pages does your suggestion apply?**

- ~~https://pub.dev/packages/audio_session#ios-setup~~
- https://pub.dev/packages/just_audio

**Quote the sentences(s) from the documentation to be improved (if any)**

> To do so, edit your ios/Podfile as follows:

**Describe your suggestion**

Update to "To do so, edit your ios/Podfile as follows AND THEN RUN POD INSTALL" ... for anyone who might not understand that that is required in order for the edit to make a difference. Not me, other people. Yup.

**Comment by ryanheise** [comment]
Thanks for the suggestion.

> **To which pages does your suggestion apply?**
> 
>     * https://pub.dev/packages/audio_session#ios-setup

That link is from audio_session which has its own issues page (can you submit one there?), but it could equally apply to just_audio which requires the same instructions.

**Comment by Zurigan** [comment]
> That link is from audio_session which has its own issues page (can you submit one there?), but it could equally apply to just_audio which requires the same instructions.

Oops, not sure how I ended up on that page ... have edited this issue, added one to the audio_session repo.

**Comment by cool2apps** [comment]
Is this documentation still valid?

I added following to Podfile as documentation described:

```
 target.build_configurations.each do |config|
      config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] ||= [
        '$(inherited)',
        'AUDIO_SESSION_MICROPHONE=0'
      ]
    end
```

But my app still asks microphone permission. Apple may reject my app just because of this, because my app does not need microphone and I can't explain them why my app requests this permission.

Also same problem happened to android side. I used this plugin for a long time on android and it did not ask for microphone recording permission. Yesterday I uploaded an update to one of my android apps, Google Play warns that your app asks microphone recording permission, this may prevent users installing your app. I did not find any documentation how I can remove this permission from android.

**Comment by ryanheise** [comment]
The documentation is still valid, with the proviso that after changing the configuration file you need to clean and rebuild.

This issue is about iOS only, but if you are experiencing an issue suddenly on both iOS AND Android, you might want to look at something else you changed in your app recently.

**Comment by cool2apps** [comment]
> The documentation is still valid, with the proviso that after changing the configuration file you need to clean and rebuild.
> 
> This issue is about iOS only, but if you are experiencing an issue suddenly on both iOS AND Android, you might want to look at something else you changed in your app recently.

You are right. Another package was using microphone. I removed that package and problem solved.

**Comment by rignaneseleo** [comment]
@ryanheise I upgraded.to just_audio: ^0.10.4

As mentioned in the migration guide, I have removed this code from the Podfile:
```
target.build_configurations.each do |config|
      config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] ||= [
        '$(inherited)',
        'AUDIO_SESSION_MICROPHONE=0'
      ]
    end
```

I deployed and received this email from apple:

> ITMS-90683: Missing purpose string in Info.plist - Your app’s code references one or more APIs that access sensitive user data, or the app has one or more entitlements that permit such access. The Info.plist file for the “Runner.app” bundle should contain a NSMicrophoneUsageDescription key with a user-facing purpose string explaining clearly and completely why your app needs the data. If you’re using external libraries or SDKs, they may reference APIs that require a purpose string. While your app might not use these APIs, a purpose string is still required.

**Comment by ryanheise** [comment]
That documentation change is probably not clear enough, since it only applies when using the latest just_audio with the latest audio_session as well (this feature is provided by audio_session). I will update the README to reflect that.
